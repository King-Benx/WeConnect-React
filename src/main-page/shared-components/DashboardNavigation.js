import React from 'react';
import {Link} from 'react-router-dom';

const DashboardNavigation = ()=>{
    return (
        <div className="col-sm-3 wrapper">
            <h3 className="text-center">WeConnect</h3>
            <div className="list-group side-menu">
                <Link to="/dashboard" className="list-group-item">
                    <i className="glyphicon glyphicon-dashboard"></i> Dashboard
                </Link>
                <Link to="/create_business" className="list-group-item" id="create_business">
                    <i className="glyphicon glyphicon-plus"></i> Create Business
                </Link>
                <Link to="/owned_businesses" className="list-group-item" id="your_business">
                    <i className="glyphicon glyphicon-eye-open"></i> Your Businesses
                </Link>
                <Link to="/all_businesses" className="list-group-item" id="view_all_businesses">
                    <i className="glyphicon glyphicon-th-list"></i> View all Businesses
                </Link>
                <Link to="/dashboard" className="list-group-item">
                    <i className="glyphicon glyphicon-search"></i> Search for Business
                </Link>
            </div>
        </div>
    );
}
export default DashboardNavigation;