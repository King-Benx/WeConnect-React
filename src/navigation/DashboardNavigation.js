import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import CustomFunctions from '../custom/CustomFunctions';

const DashboardNavigation = (props)=>{
    return (
        CustomFunctions.checkAuthentication() ? ( 
        <div className="col-sm-3 wrapper">
            <h3 className="text-center">WeConnect</h3>
            <div className="list-group side-menu">
                <Link to="/dashboard" className="list-group-item">
                    <i className="glyphicon glyphicon-dashboard"></i> Dashboard
                </Link>
                <Link to="/create_business" className="list-group-item" id="create_business">
                    <i className="glyphicon glyphicon-plus"></i> Create Business
                </Link>
                <Link to="/owned_businesses" className="list-group-item" id="your_business">
                    <i className="glyphicon glyphicon-eye-open"></i> Your Businesses
                </Link>
                <Link to="/all_businesses" className="list-group-item" id="view_all_businesses">
                    <i className="glyphicon glyphicon-th-list"></i> View all Businesses
                </Link>
                <Link to="/reset_password" className="list-group-item" id="view_all_businesses">
                    <i className="glyphicon glyphicon-wrench"></i> Reset Password
                </Link>
            </div>
        </div>
        ):<Redirect to={{pathname:'/login'}}/>
    );
}
export default DashboardNavigation;