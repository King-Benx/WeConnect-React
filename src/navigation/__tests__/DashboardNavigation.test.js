import React from 'react';
import { shallow } from 'enzyme';
import DashboardNavigation from "../DashboardNavigation";
import renderer from 'react-test-renderer';

describe('These are tests for the dashboard Navigation', () => {
    let mountedDashboardNavigation;

    beforeEach(()=>{
        mountedDashboardNavigation= shallow(<DashboardNavigation />);
    });

    it('renders without crashing', () => {
        shallow(<DashboardNavigation />);
        const tree = renderer.create(<DashboardNavigation />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has a redirect', () =>{
        const lists = mountedDashboardNavigation.find('Redirect')
        expect(lists.length).toBe(1)
    });
    
});