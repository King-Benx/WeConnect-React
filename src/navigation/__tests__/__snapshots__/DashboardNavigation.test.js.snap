// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the dashboard Navigation renders without crashing 1`] = `null`;
